### DISCUSSION TITLE
Command Component. How to make the CommandList overlap the content outside (like google search)

### DISCUSSION THREAD

**Discussion by cmgchess** [shadcn-ui/ui#2808] [Q&A]
I'm using the Command component. My goal is to use it as an Autocomplete component. However, I am facing an issue where the content below the Command component adjusts its position based on the dropdown of search results. Ideally, the search result dropdown should overlap the content below, similar to how it works in Google Search.

Here is the demo  
[Stackblitz](https://stackblitz.com/edit/stackblitz-starters-sa2vwv?file=app%2Fpage.tsx)

![image](https://github.com/shadcn-ui/ui/assets/61736812/e49afd04-3c30-47a8-9852-4595a1efdbe3)
![image](https://github.com/shadcn-ui/ui/assets/61736812/0f0864dd-adaf-4ed1-8e10-3281439e340f)
![image](https://github.com/shadcn-ui/ui/assets/61736812/a055af44-4f1a-49a7-8a5f-de800ae4ebdf)

Thanks in advance 🙏

**Comment by AndreaBarghigiani** [comment]
I've been in your same boat, spent few hours searching and testing few implementations but in the end it never worked out properly.

If I spent more time on it, maybe, I could've been able to solve all the issues and have a proper autocomplete. But I decided to move over and find something that was built just to create an autocomplete experience.

Lucky me I remembered that [`downshift-js`](https://www.downshift-js.com/) does **exactly that**, and you can reuse all the beautiful components of shadcn/ui because it exposes the `useCombobox` Hook that gives you all the control you want 😉

**Comment by lukebelbina** [comment]
Did you ever figure this out?

**Comment by theodor349** [comment]
I have the same problem, and have been trying to solve it using the `Popover` component (Still not solved though).
Something like this:
```tsx
const [open, setOpen] = React.useState(false);

<Command>
  <CommandInput onFocus={() => {setOpen(true)}}/>
  <Popover open={open} onOpenChange={setOpen}>
    <PopoverTrigger>
    </PopoverTrigger>
    <PopoverContent>
      <CommandGroup>
        <CommandList>
          {selection.map((value) => (
            <CommandItem
              key={value}
              value={value}
              onSelect={(currentValue) => {
                setOpen(false);
              }}
            >
              {value}
            </CommandItem>
          ))}
        </CommandList>
      </CommandGroup>
    </PopoverContent>
  </Popover>
</Command>

```

**Comment by AndreaBarghigiani** [comment]
As I stated previously https://github.com/shadcn-ui/ui/discussions/2808#discussioncomment-8546453, I solved my issue with [downshift-js](https://downshift-js.com/) but TBH it never felt that I was following the shadcn/ui approach 😅

Maybe you can have a look at this [fancy multiselect](https://craft.mxkaske.dev/post/fancy-multi-select).

PS: also have a look at this [great repo](https://github.com/birobirobiro/awesome-shadcn-ui) full of awesome components made from and for shadcn/ui. The repo is not mine, but I think we should spread it more because it's full of interesting components build from various developers. It's an Awesome List after all
